spec/system/candidate_interface/candidate_signed_in_user_with_course_params_spec.rb

```diff
@@ -0,0 +1,173 @@
+require 'rails_helper'
+
+RSpec.describe 'An existing candidate arriving from Find with a course and provider code' do
+  include CourseOptionHelpers
+  scenario 'candidate is signed in' do
+    given_the_pilot_is_open
+    and_i_am_an_existing_candidate_on_apply
+    and_i_have_less_than_3_application_options
+    and_the_course_i_selected_only_has_one_site
+    and_i_am_signed_in
+
+    when_i_arrive_at_the_apply_from_find_page_with_course_params_with_one_site
+    and_i_click_apply_on_apply
+    then_i_should_see_the_courses_review_page
+    and_i_should_see_the_course_name_and_code
+    and_i_should_see_the_site
+
+    when_i_arrive_at_the_apply_from_find_page_with_the_same_course_params
+    and_i_click_apply_on_apply
+    then_i_should_see_the_courses_review_page
+    and_i_should_be_informed_i_have_already_selected_that_course
+
+    given_the_course_i_selected_has_multiple_sites
+    and_i_am_signed_in
+    and_i_have_less_than_3_application_options
+
+    when_i_arrive_at_the_apply_from_find_page_with_course_params_with_multiple_sites
+    and_i_click_apply_on_apply
+    then_i_should_see_the_course_choices_site_page
+    and_i_see_the_form_to_pick_a_location
+
+    given_the_course_i_selected_has_multiple_sites
+    and_i_am_an_existing_candidate_on_apply
```

Comment:
Creating a new candidate in the middle of the test confused me a bit. But I think we should probably leave as is - we don't want to break this up and multiply the number of tests we have. What do you think?

Reply:
Yea I get that and it definitely could be split over more test files. I agree though that the overhead of creating the extra files and duplicating the majority of the methods probably isn't worth it.